### DISCUSSION TITLE
Alpha in capillaryPressureVG as a function of temperature

### DISCUSSION THREAD

**Discussion by mcacace** [idaholab/moose#16735] [Q&A Modules: General] [answered]
This post is on the porous flow module. 
I am trying to help a colleague of mine to implement a VG retention curve where the alpha coefficient is not considered as constant but varies with the background temperature condition. Looking at the respective uo, it does seem to me (first glance though) that the basic virtual functions can retrieve this information via the corresponding qp during their call. This said, being my first attempt with the module was wondering what would be the best way to implement such functional behaviour.
Thanks for helping, 
Mauro

**Comment by cpgr** [comment]
Hi Mauro,

Without thinking about it too much, one way might be to change the param `alpha` to a coupled variable, and then use an auxvariable to compute the temperature-dependent value of `alpha`. Like you said, you could then pass `_alpha[_qp]` into the various methods in this class.

Let me know if you need a hand,
Chris

**Reply by mcacace** [reply]
Dear Chris,
thanks for the prompt reply. Was thinking the same, but wanted to be sure I was heading the right direction. Will get at it and let you know if I have any problem.

**Comment by fparisio-zz** [comment]
Dear Chris,

I am working with Mauro and together we are trying to explore the role of capillary forces in water-vapor systems. I have built a first simple example employing PorousFlowFluidStateSingleComponent where primary variables are pliq and h (files attached). The model is a single element and I am fixing enthalpy and pressure in order to follow an evaporation isotherm (here, 200 °C). Once the whole fluid has evaporated and liquid saturation is null, a full gas phase forms at a gas pressure which is controlled by the retention curve properties. Because the upper limit of pressure in the water EOS is 100 MPa, the maximum pressure has to be fixed accordingly, to avoid that the gas pressure increases to values that are above such limits (for our cases, approx. pc_max=70 MPa). Now, we observe that once the water has fully evaporated and the gas saturation is 1, the pressure to compute EOS properties is suddenly taken as the gas pressure (pliq is meaningless with fully gaseous phase) and this results in a jump in temperature up to approximately 550 °C. Now, this temperature is the result of the fixed value of enthalpy and pressure once the liquid phase disappears. Because of the shape and parametrization of the retention curve, this jump is more pronounced for a material that shows higher retention capabilities or, in other words, it is proportional to pc_max (see figure with different cases of pc_max). I understand this is how VG works (or any other retention model), but in reality, once the liquid phase has disappeared, the capillary pressure should be null and pgas=pliq at sgas=1. One way around could be achieved by a residual saturation that prevents a full evaporation (sgas<1 always), but that case implies higher values of capillary pressure and, as a consequence, gas pressure exceeds the threshold pgas>100 MPa and EOS cannot be computed. 

At present, we are not sure what would be the best way out of this loop and we would like to ask you if you have any suggestion for us. In the examples provided in PorousFlow, the maximum capillary pressure is usually rather low (around 1 MPa) and the problem does not arise (or, the temperature jump is rather small). We could also discuss about the fact that perhaps retention properties of rocks are overestimated: the colored experimental retention curves (from Li and Horne, 2006) in figure d are experiments from reservoir rocks obtained with Mercury Porosimetry Intrusion, which might not be completely representative of water-vapor systems. Especially at high temperature, the lower surface tension would cause a decrease of retention capabilities of a porous medium and a lower pc_max (hence, our idea of a temperature-dependent alpha in the VG curve). Nonetheless, even with reduced retention capabilities (tens of MPa of pc_max), the current formulation falls short (although we understand it is an intrinsic problem of the retention theory).

Thank you very much for your help,

Francesco

Ref.:
Li, K. and Horne, R.N., 2006. Fractal modeling of capillary pressure curves for The Geysers rocks. Geothermics, 35(2), pp.198-207.

![evaporation_isotherms](https://user-images.githubusercontent.com/23402125/105210632-970fc500-5b4b-11eb-9ac7-1283c2b84fe4.png)

[moose_input.tar.gz](https://github.com/idaholab/moose/files/5844093/moose_input.tar.gz)

**Reply by fparisio-zz** [reply]
PS: we can also discuss this matter off-line if it requires longer replies and if you are interested in discussing more about the details.

**Reply by WilkAndy** [reply]
No, let's keep this online - it might be useful for someone in the future

**Comment by WilkAndy** [comment, marked as answer]
I can't make any better suggestions than what's already been said (all your suggestions are potentially good ones, i think), but just wanted to say how exciting it is to see PorousFlow being used in this way.  When we created it, we tried to make it flexible enough that it'd be able to handle cases such as this, and i'm so pleased it's working for you.

My uneducated opinion is that you're using VG in situations where it's incorrect.  I don't think i've seen experimental literature at such high temperatures and low liquid saturations.

Also, just for my interest, what's the problem with the behaviour at zero liquid saturation?  Are you trying to boil water and then do something with it afterwards?


a

**Reply by fparisio-zz** [reply]
@WilkAndy your opinion is actually very well educated! I agree that experimental evidence lacks at such low saturations and that it is not straight forward to extend retention curves to water-vapor systems. Perhaps the most informative study in this sense is by Li and Horne (2007).

The test simulated has no a particular significance other than exploring the capillary forces that develop during an evaporation isotherm. It will be functional to a more systematic study. Thank you for your input and we will post here whevener we find a proper way to handle the problem.

Francesco 

Ref:
Li, K. and Horne, R.N., 2007. Systematic study of steam–water capillary pressure. Geothermics, 36(6), pp.558-574.

**Comment by cpgr** [comment]
Interesting work!

I'm not sure what the best step is, but will think about it today. From memory, the temperature is calculated from pressure and enthalpy using the IAPWS97 equation, and it has something like this:

```
switch(phase):

case liquid:
  T = T_from_p_h(liq_p, h)

case gas:
  T = T_from_p_h(gas_p, h)
```

which could cause the sudden jump in temperature that you observe.

Chris

**Reply by fparisio-zz** [reply]
So it seems, that a switch occurs whenever the liquid phase disappears. The effect of capillary pressure is to introduce a discontinuity in the pressure at full evaporation, which then translates into a temperature jump.
